text/0000-windows-subsystem.md

```diff
@@ -0,0 +1,161 @@
+- Feature Name: Windows Subsystem
+- Start Date: 2016-07-03
+- RFC PR: ____
+- Rust Issue: ____
+
+# Summary
+[summary]: #summary
+
+Rust programs compiled for windows will always allocate a console window on
+startup. This behavior is controlled via the `SUBSYSTEM` parameter passed to the
+linker, and so *can* be overridden with specific compiler flags. However, doing
+so will bypass the rust-specific initialization code in `libstd`, as when using
+the MSVC toolchain, the entry point must be named `WinMain`.
+
+This RFC proposes supporting this case explicitly, allowing `libstd` to
+continue to be initialized correctly.
+
+# Motivation
+[motivation]: #motivation
+
+The `WINDOWS` subsystem is commonly used on windows: desktop applications
+typically do not want to flash up a console window on startup.
```

Comment:
This is more about what the user sees rather than a technical detail: ie. console applications can hide the console, but you can't stop it flashing up momentarily. The motivation being to improve the user experience.